openghg/store/_obssurface.py

```diff
@@ -384,14 +364,17 @@ def read_file(
                 for key, value in data.items():
                     data[key]["data"] = value["data"].chunk(chunks)
 
-            # Mop up and add additional keys to metadata which weren't passed to the parser
-            data = self.update_metadata(data, additional_metadata)
+            # Check to ensure no required keys are being passed through optional_metadata dict
```

Comment:
Lines 367-377 look like they implement roughly the same logic as before, but it's not obvious that they do. Refactoring `get_lookup_keys` should probably be a different PR. To pass metadata around the parser, it seems like it would be enough to merge `additional_input_parameters` into the data that comes out of the parser, then continue as before.